apps/examples/binary_update/binary_update_main.c

```diff
@@ -78,20 +110,130 @@ static void binary_update_getinfo_all(void)
 	}
 }
 
-static void binary_update_getinfo(char *name)
+static void binary_update_info(binary_info_t *pre_bin_info, binary_info_t *cur_bin_info, int condition)
+{
+	printf(" === [%4s] Update info === \n", cur_bin_info->name);
+	printf(" %4s | %8s | %s \n", "Con", "Version", "Inactive part");
+	printf(" ------------------------- \n");
+	printf(" %4s | %8s | %s \n", "Pre", pre_bin_info->version, pre_bin_info->dev_path);
+	printf(" %4s | %8s | %s \n", "Cur", cur_bin_info->version, cur_bin_info->dev_path);
+	printf(" ========================= \n");
+
+	if (condition == DOWNLOAD_SUCCESS) {
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) != 0)) {
+			fail_cnt++;
+			printf("Binary Update Download Success Case FAIL.\n");
+		} else {
+			printf("Binary Update Download Success Case SUCCESS.\n");
+		}
+	} else { //DOWNLOAD_FAIL
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) == 0)) {
+			fail_cnt++;
+			printf("Binary Update Download Fail Case FAIL.\n");
+		} else {
+			printf("Binary Update Download Fail Case SUCCESS.\n");
+		}
+	}
+}
+
+static void binary_update_download(char devname[BINMGR_DEVNAME_LEN], int condition)
+{
+	int fd;
+	int ret;
+	int file_size;
+	int read_size;
+	uint32_t crc_hash;
+	uint8_t crc_buffer[CRC_BUFFER_SIZE];
+	binary_header_t header_data;
+
+	printf("\n** Binary Update Download [%s] %s case test.\n", devname, condition == DOWNLOAD_SUCCESS ? "Success" : "Fail");
+
+	fd = open(devname, O_RDONLY);
+	if (fd < 0) {
+		printf("Failed to open %s: %d\n", devname, ret);
+		fail_cnt++;
+		return;
+	}
+
+	/* Read the checksum */
+	ret = read(fd, (FAR uint8_t *)&crc_hash, CHECKSUM_SIZE);
+	if (ret != CHECKSUM_SIZE) {
+		printf("Failed to read %s: %d\n", devname, ret);
+		fail_cnt++;
+		return;
```

Comment:
In this case, you'd better close the fd.

Reply:
good! thanks